src/sdk/model/vm/guest-tools/guest-tools.ts

```diff
@@ -0,0 +1,60 @@
+import { GuestToolsJson } from './__json__';
+
+/**
+ * VM Guest Tools
+ */
+export class GuestTools {
+  private _json: GuestToolsJson;
+
+  constructor(guestTools: GuestTools);
+  constructor(guestToolsJson: GuestToolsJson);
+  constructor(firstParam: boolean | GuestTools | GuestToolsJson) {
+    if (firstParam instanceof GuestTools) {
+      // Copy constructor
+      this._json = (firstParam as GuestTools).json;
+    } else {
+      // Json or empty constructor
+      this._json = (firstParam || {}) as GuestToolsJson;
+    }
+  }
+
+  /**
+   * Returns guest tools status
+   * @returns {string}
+   */
+  get status(): string {
```

Comment:
Same here, if this._json is empty, this method won't return a string, but undefined instead.